### PR TITLE
Add missing virtual destructor

### PR DESCRIPTION
**Pull request by tautschnig**

### CODE REVIEW

**Review by smowton — APPROVED** [review]

**Comment by kroening** [comment]
On some of these I wonder whether the virtual member function really needs to be virtual. The class could likely be better final.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: a6324b171d1e18ed084fc77c3da6e6bb947bef47).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/77134438

File: src/util/endianness_map.h [hunk 1]
```diff
@@ -41,6 +41,8 @@ class endianness_mapt
   {
   }
 
+  virtual ~endianness_mapt() = default;
```
Comment (peterschrammel) [inline src/util/endianness_map.h:44]: I agree with @kroening that some classes should rather be made final, such as this one.

File: jbmc/src/java_bytecode/java_bytecode_parse_tree.h [hunk 2]
```diff
@@ -84,6 +84,8 @@ class java_bytecode_parse_treet
       is_private(false), is_static(false), is_final(false)
     {
     }
+
+    virtual ~membert() = default;
```
Comment (peterschrammel) [inline jbmc/src/java_bytecode/java_bytecode_parse_tree.h]: make final?

File: src/solvers/floatbv/float_bv.h [hunk 3]
```diff
@@ -22,9 +22,7 @@ class float_bvt
   {
   }
 
-  ~float_bvt()
-  {
-  }
+  virtual ~float_bvt() = default;
```
Comment (peterschrammel) [inline src/solvers/floatbv/float_bv.h]: make final?

**Review by peterschrammel — APPROVED** [review]

File: jbmc/src/java_bytecode/java_bytecode_parse_tree.h (hunk 2 above)
Comment (tautschnig) [reply jbmc/src/java_bytecode/java_bytecode_parse_tree.h]: But ... this class has children deriving from it?

Comment (tautschnig) [reply jbmc/src/java_bytecode/java_bytecode_parse_tree.h]: Well, there is case to be made for private inheritance, the use of `virtual` indeed does not seem to be justified.

**Comment by tautschnig** [comment]
@peterschrammel @kroening I have reduced this patch down to the minimum of those that really need to be `virtual`. I will create separate PRs to clean up all the others, and will merge this one once CI passes.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 0c9658f988d5d5725f6906df63c0158fdb858802).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/78382838

**Pull request merged**